### PR TITLE
All Mutation Tox acts like old Slime Toxin; Old Slime Toxin merged with (Classic) Mutation Toxin

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
## About The Pull Request

Replaces the instant race-changing effect with a system currently used by Slime Mutation Toxin.

This means you must undergo 20 iterations of the chem before it changes you.

I made the messages generic because cba, you can add unique ones though!

## Why It's Good For The Game

Allows for some counterplay if you get injected with a bad kind (aka PURGE IT PURGE IT) while also requiring a hint of prep should you need to use the toxin advantageously. Definitely more likely to fall into the former rather than the latter.

## Changelog
:cl: Mutationby (Idea by Arathian)
tweak: Mutation tox now has a minute window where you can purge the chem to avoid being changed (It functions EXACTLY like old slime mutation toxin)
tweak: Slime Mutation Toxin is replaced by Mutation Toxin since they now function 100% the same. Think of this as just a name change!
/:cl:

### CODE REVIEW

**Comment by Arathianen** [comment]
That works with all mutation toxins, right?

So lizard mutation toxin, plasmaman mutation toxin, felinid etc, right?

**Comment by ExcessiveUseOfCobblestone** [comment]
yep

**Comment by FantasticFwoosh** [comment]
But now we have to wait for 20 to turn back with human (or insert relevant) mutation toxin which was the cure anyway; I disapprove of this anti QoL PR.

**Comment by Krysonism** [comment]
Can we lower the metabolism rate? If you find a maint pill with a cool toxin you want to be able to split it into smaller doses.

**Comment by ExcessiveUseOfCobblestone** [comment]
I can't lower a 0 metabolization rate

**Comment by ExcessiveUseOfCobblestone** [comment]
> But now we have to wait for 20 to turn back with human (or insert relevant) mutation toxin which was the cure anyway; I disapprove of this anti QoL PR.

which is problematic? If you get turned you should have a window where the use can prevent reconversion (counterplay) besides playing hot potato with 2 different mut tox's

**Comment by Shadowflame909** [comment]
Does this effect slime person toxin? It has a gimmick where it takes a while to first use. But will generally be faster the next times because you'll reroll the same type 5 times in a row and waste a lot of slimes.

**Comment by ATH1909** [comment]
> Does this effect slime person toxin? It has a gimmick where it takes a while to first use. But will generally be faster the next times because you'll reroll the same type 5 times in a row and waste a lot of slimes.

"tweak: Slime Mutation Toxin is replaced by Mutation Toxin since they now function 100% the same. Think of this as just a name change!"

**Comment by Shadowflame909** [comment]
Fair. It just has two functions  though. Or rather a speedier version of its first

**Comment by ExcessiveUseOfCobblestone** [comment]
The speedier version is now the "slow" version and the mechanic of being "slow" has been applied to all mutation toxins.

File: code/modules/reagents/chemistry/reagents/other_reagents.dm [hunk 1]
```diff
@@ -427,178 +427,177 @@
...
+	taste_description = "grandma's gelatin"
+
+/datum/reagent/mutationtoxin/jelly/on_mob_life(mob/living/carbon/human/H)
+	if(isjellyperson(H))
+		to_chat(H, "<span class='warning'>Your jelly shifts and morphs, turning you into another subspecies!</span>")
+		var/species_type = pick(subtypesof(/datum/species/jelly))
+		H.set_species(species_type)
+		H.reagents.del_reagent(type)
+		return TRUE
+	if(current_cycle >= 20) //overwrite since we want subtypes of jelly
```
Comment (XDTM) [inline code/modules/reagents/chemistry/reagents/other_reagents.dm]: This'll stop working if someone decides to only lower the default threshold, it should probably be moved to a function to override or at least a variable

**Comment by Ty-the-Smonk** [comment]
Aw no more stun toxin

**Comment by Shadowflame909** [comment]
QOL tanked for the green slime xenobiologist

**Comment by optimumtact** [comment]
xenobios have had it good for years

**Comment by ExcessiveUseOfCobblestone** [comment]
it's not really QoL if it allows for direct weaponization lol...

**Comment by Shadowflame909** [comment]
one of the rare few oranges

still talking about the slime person re-roll RNG that honestly is very unlucky and very wasteful

**Comment by ExcessiveUseOfCobblestone** [comment]
jellyswitching was retained, it doesn't require 20 iterations each.

**Pull request merged**